File: CPP/grassland.cpp
#include <iostream>
using namespace std ;

#include "grassland.h"

void Grassland::render() {
	if (!this->getCreation() ){
		if (getGrass() ){
			cout << "+" ;
		}else{
			cout << "-" ;
		}
	}
	else
	{
		this->getCreation()->render() ;	
	}
	
}

int Grassland::getIdentity() {
	return 2 ;
}

Grassland::Grassland() {
	this->setGrass(false) ;
}

Grassland::Grassland(bool _r) {
	this->setGrass(_r) ;
}

File: main/land/Grassland.java
package main.land;
//#include "grassland.h"
//#include "land.h"

public class Grassland extends Land{

public void render() {
	if (this.getCreation() == null){
		if (getGrass() ){
			System.out.print("+") ;
		}else{
			System.out.print("-") ;
		}
	}
	else
	{
		this.getCreation().render() ;	
	}
	
}

public int getIdentity() {
	return 2 ;
}

public Grassland() {
	this.setGrass(false) ;
}

public Grassland(boolean _r) {
	this.setGrass(_r) ;
}

}
